sandbox/apparmor/notify/permission.go

```diff
@@ -32,9 +32,9 @@ const (
 	AA_MAY_OPEN
 	// AA_MAY_RENAME implies that a process may rename a file.
 	AA_MAY_RENAME
-	// AA_MAY_SETATTR is not checked by the kernel.
+	// AA_MAY_SETATTR is not checked by the kernel but may appear in messages.
```

Comment:
it is checked by the kernel. Its just the kernel or the compiler is mapping the write permission to include it. It a rule that covers it doesn't exist, or a prompt rule covers it then you will actually get the actually permission requested by the kernel in the message.

Reply:
So if there's a rule in place, it is treated as `AA_MAY_WRITE`? Or it is handled if there's a `w` rule?

I'm a bit confused as to the distinction between `AA_MAY_WRITE`, `AA_MAY_APPEND` (for example), and `AA_MAY_SETATTR`. What does it mean that the latter is not exposed to the user in policy?

Reply:
Or is `AA_MAY_[SG]ETATTR` more like `AA_MAY_OPEN`, where it's sometime included with one of `AA_MAY_READ`, `AA_MAY_WRITE`, etc., but not directly mapped to `r`, `w`, `x`?

Reply:
on the policy side append is treated as a subset of write. You can specify append permission without giving full write access, but write will grant append as well as write. Kernel side they are separated out sort of. That is if a file is opened for appending writes the check will only be for append, but if the file is opened for write, the permission check will be against write even if the write is only ever in position that appends to the file.

> Or is AA_MAY_[SG]ETATTR more like AA_MAY_OPEN, where it's sometime included with one of AA_MAY_READ, AA_MAY_WRITE, etc., but not directly mapped to r, w, x?

newer abis allow all of them to be specified separately. However atm policy does not allow this for files, however some of them are possible to specified separately with other mediation types. The plan is to make the fine grained permissions available in policy at some point while retaining backwards compatibility. That is 'w' for files will always map to multiple permissions but in the future something like (syntax not finalized) "setattr" will be available to specify the individual permissions.